src/gpuarray_sort.c

```diff
@@ -0,0 +1,1310 @@
+#include <assert.h>
+#include <limits.h>
+#include <float.h>
+
+#include <gpuarray/sort.h>
+#include <gpuarray/array.h>
+#include <gpuarray/util.h>
+
+#include "util/strb.h"
+#include "private.h"
+
+/*
+ * Copyright 1993-2015 NVIDIA Corporation.  All rights reserved.
+ *
+ * Please refer to the NVIDIA end user license agreement (EULA) associated
+ * with this source code for terms and conditions that govern your use of
+ * this software. Any use, reproduction, disclosure, or distribution of
+ * this software and related documentation outside the terms of the EULA
+ * is strictly prohibited.
```

Comment:
It feels a bit weird to refer to an EULA that is not there.  Perhaps there is a way to reword that.